packages/lodestar/src/util/httpClient.ts

```diff
@@ -0,0 +1,62 @@
+import Axios, {AxiosResponse, AxiosInstance} from "axios";
+import { ILogger } from "../logger";
+
+export interface HttpClientOptions {
+  // Add more options if needed
+  urlPrefix: string;
+}
+
+export class HttpClient {
+  private client: AxiosInstance;
+  private logger: ILogger;
+
+  public constructor(opt: Partial<HttpClientOptions>, logger: ILogger) {
+    this.client = Axios.create({
+      baseURL: opt.urlPrefix || ""
+    });
+    this.logger = logger;
+  }
+
+  public get<T>(url: string): Promise<T> {
+    return new Promise(async (resolve, reject) => {
```

Comment:
you don't need promise here, you can just return data and in handleError do `throw new Error(<custom message>)`.
Later on, we would like to throw some custom exception classes